### PR TITLE
Graphql api docs

### PR DESCRIPTION
**Pull request by alvaroloes**
Improves de documentation of the Auth API, removes the old documentation about the REST API, and adds documentation about the GraphQL API

### CODE REVIEW

File: source/includes/_application-api.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
+# Your application API
+
+The API is the way the outside world interact with your system, and any server-side application needs one.
+With Booster, you don't need to worry about it: _it is created for you_ and evolves with your code _automatically_.
```
Comment (NickSeagull) [inline source/includes/_application-api.md]: It's just my opinion, but it doesn't look like this sentence is saying anything useful to me. It's trying to sell me something, although I'm already reading the docs. I'd rephrase it as:

```suggestion
The API is generated automatically when you deploy your application, and it changes when needed, no additional steps required on your side.
```

File: source/includes/_application-api.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+When you finish deploying your application using the `boost deploy` command, it will print useful information needed to 
+use your API. It will be shown under the "Outputs" section.
+
+```sh
+Outputs:
+<application name>.httpURL = https://<API ID>.execute-api.<region>.amazonaws.com/<environment name>
+<application name>.websocketURL = wss://<API ID>.execute-api.<region>.amazonaws.com/<environment name>
+<application name>.clientID = abcdXjk1234
+```
```
Comment (NickSeagull) [inline source/includes/_application-api.md:13]: Note that this is explaining how it works in AWS. Another provider might not have this code. I'd remove this and say that after the application is deployed, the endpoints will be shown

File: source/includes/_application-api.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+    "__type": "InvalidParameterException",
+    "message": "Username should be an email."
+}
+```
+
+You will get an HTTP status code different from 2XX and a body with a message telling you the reason of the error.
+
+### Sign-in
+Allows your users to get tokens to be able to make request to restricted endpoints.
+Remember that before a user can be signed in into your application, **its email must be confirmed**
```
Comment (NickSeagull) [inline source/includes/_application-api.md]:
```suggestion
Remember that before a user can be signed in into your application, **their email must be confirmed**
```

File: source/includes/_application-api.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+	"clientId":"string",
+	"username":"string",
+	"password":"string"
+}
+```
+
+Parameter | Description
+--------- | -----------
+_clientId_ | The application client Id that you got as an output when the application was deployed.
+_username_ | The username of the user you want to sign in. It must be previously signed up
```
Comment (NickSeagull) [inline source/includes/_application-api.md]:
```suggestion
_username_ | The username of the user you want to sign in. They must have previously signed up
```

File: source/includes/_application-api.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+
+## Commands and ReadModels API
+ 
+This is the main API of your application, as it allows you to:
+
+ - _Modify_ data by **sending commands**
+ - _Read_ data by **querying read models**
+ - _Receive data in real time_ by **subscribing to read models** 
+ 
+All this is done through [GraphQL](https://graphql.org/), a nice query language for APIs that has useful advantages over simple REST APIs.
```
Comment (NickSeagull) [inline source/includes/_application-api.md]: Let's make docs un-opinionated

```suggestion
All this is done through [GraphQL](https://graphql.org/), a query language for APIs that has useful advantages over simple REST APIs.
```

**Review by NickSeagull — CHANGES_REQUESTED** [review]
I'd extract the auth part into a separate file and call it `Auth in AWS` or something like that

File: source/includes/_application-api.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+```sh
+Outputs:
+<application name>.httpURL = https://<API ID>.execute-api.<region>.amazonaws.com/<environment name>
+<application name>.websocketURL = wss://<API ID>.execute-api.<region>.amazonaws.com/<environment name>
+<application name>.clientID = abcdXjk1234
+```
+
+The meaning of those values are:
+
+- **httpURL**: This is the main URL of your application. You will need it to interacting with the
```
Comment (javiertoledo) [inline source/includes/_application-api.md]:
```suggestion
- **httpURL**: This is the base HTTP URL of your application. You will need it to interacting with the 
```

File: source/includes/_application-api.md [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+        },
+        {
+          "sku": "ABC_02",
+          "quantity": 3
+        }
+      ]
+    }
+  }
+}
+```
```
Comment (javiertoledo) [inline source/includes/_application-api.md:452]: I think that it could be a good idea to show how to do this with ApolloClient in addition to the raw requests to show more realistic usage beyond pure documentation. We can do that in the new doc pages, no need to add it in this PR. /cc @NickSeagull @itrion

**Review by javiertoledo — APPROVED** [review]
Let's merge this into the new docs, we'll keep editing them there

File: source/includes/_application-api.md (hunk 7 above)
Comment (alvaroloes) [reply source/includes/_application-api.md:452]: Totally. I left that for the future because we still need to ensure Booster is compatible with the subscriptions-transport-ws to get full compatibility with Apollo

Comment (alvaroloes) [reply source/includes/_application-api.md:452]: BTW, feel free to merge the PR if you need that!

File: source/includes/_application-api.md (hunk 2 above)
Comment (alvaroloes) [reply source/includes/_application-api.md:13]: I agree, but let's do it in another PR in the new docs, just to merge this and move the docs there
